src/streaming.rs

```diff
@@ -0,0 +1,270 @@
+use ::open_vcdiff_sys;
+
+use std::error;
+use std::fmt;
+use std::marker::PhantomData;
+use std::mem;
+use std::os::raw::c_void;
+use std::slice;
+use std::result;
+
+#[derive(Debug,Copy,Clone,PartialEq,Eq)]
+pub struct Error(&'static str);
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        self.0
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "vcdiff error: {}", self.0)
+    }
+}
+
+pub type Result<T> = result::Result<T,Error>;
+
+pub trait Output {
```

Comment:
Not sure if it's a good idea (because of exception handling issues) but maybe `Output` could be implemented in terms of `Write` somehow?